### ISSUE TITLE
[BUG] libcudf.so crash on open with "basic_string::_S_construct null not valid"

### ISSUE THREAD

**Issue by jlowe** [rapidsai/cudf#3722]
**Describe the bug**
Loading libcudf.so outside of a conda environment can crash with the error:
```
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_S_construct null not valid
```

**Steps/Code to reproduce bug**
Removing LIBCUDF_INCLUDE_DIR and  CONDA_PREFIX from the environment will trigger it.  For example:
```
$ env -u CONDA_PREFIX -u LIBCUDF_INCLUDE_DIR ./gtests/LEGACY_BITMASK_TEST
terminate called after throwing an instance of 'std::logic_error'
  what():  basic_string::_M_construct null not valid
Aborted (core dumped)
```

**Expected behavior**
libcudf should not crash on open.  If for some reason it must do so the error message should be very clear to the user what is the cause of the crash.

**Comment by jlowe** [comment]
I believe this is caused by the following code in binaryop.cpp:
```c++
#ifndef LIBCUDF_INCLUDE_DIR
#define LIBCUDF_INCLUDE_DIR \
  std::string{std::getenv("CONDA_PREFIX")} + "/include/libcudf"
#endif

// env var always overrides the default value of LIBCUDF_INCLUDE_DIR
const char* env_dir = std::getenv("LIBCUDF_INCLUDE_DIR");
const std::string libcudf_include_dir =
    env_dir != NULL ? env_dir : LIBCUDF_INCLUDE_DIR;
```

If LIBCUDF_INCLUDE_DIR is not set it will try to fallback to CONDA_PREFIX.  However if CONDA_PREFIX is also not set, `getenv` will return NULL which is illegal for a string constructor.

**Comment by kkraus14** [comment]
ping @trxcllnt

**Comment by jlowe** [comment]
Fixed by #3699
